#!/usr/bin/env python
PACKAGE = "breadcrumb"

from dynamic_reconfigure.parameter_generator_catkin import *
import math

gen = ParameterGenerator()

heuristic_enum = gen.enum( [ gen.const("Manhattan", int_t, 0, "Sets A* to use the Manhattan heuristic"),
                             gen.const("Euclidean", int_t, 1, "Sets A* to use the Euclidean heuristic"),
                             gen.const("Octagonal", int_t, 2, "Sets A* to use the Octagonal heuristic")],
                           "An enum representing the search heuristic")

gen.add("allow_diagonals", bool_t, 0, "Allows the path tracking to output diagonal movements", True)
gen.add("obstacle_threshold", int_t, 0, "At what occupancy probability where an obstacle is counted in the search", 50, 0, 100)
gen.add("search_heuristic", int_t, 0, "Heuristic to use during A* search", 1, 0, 2, edit_method=heuristic_enum)
gen.add("calc_sparse_path", bool_t, 0, "Enables calculation of the sparse path result", True)

exit(gen.generate(PACKAGE, "breadcrumb", "AStarParams"))
